src/datasets/arrow_dataset.py

```diff
@@ -2653,6 +2653,20 @@ def add_elasticsearch_index(
             )
         return self
 
+    def add_item(self, item: dict):
+        """Add item to Dataset.
+
+        Args:
+            item (dict): Item data to be added.
+        """
+        table = InMemoryTable.from_pydict({k: [v] for k, v in item.items()})
+        # Cast item
+        type = self.features.type
+        schema = pa.schema({col_name: type[col_name].type for col_name in self._data.column_names})
+        table = table.cast(schema)
+        # Concatenate tables
+        self._data = concat_tables([self._data, table])
```

Comment:
Maybe here I could use `ConcatenationTable.from_tables` instead.

Reply:
Or even better, `ConcatenationTable.from_blocks`.

Reply:
Either `concat_tables` or `ConcatenationTable.from_tables` are fine :)

But `ConcatenationTable.from_blocks` only takes `InMemoryTable` or `MemoryMappedTable` objects as input so it may fail if `self._data` is already a `ConcatenationTable`

Reply:
Good. I leave as it is then.